lib/checks/navigation/region-after.js

```diff
@@ -0,0 +1,30 @@
+import { matchAncestry } from '../../core/utils';
+
+function regionAfter(results) {
+  results.forEach((r, index) => {
+    // this element failed the check
+    if (!r.result) {
+      // we need to see if it's actually contained in an iframe that passed the check
+      const ancestryMinusLast = r.node.ancestry.slice(0, -1);
+
+      // we only need to check up to index - 1 because the iframe this element might be contained in will have been found before this element
+      for (const earlierResult of results.slice(0, index)) {
```

Comment:
This feels unnecessarily complex. I suggest you create a list of frames with `result:true`, and then try to match against that. You have no business running `matchAncestry` against anything other than iframes. You don't need to do it if ancestry has a length of 1 either.

That also lets you use clearer variable names. It took me a while to figure out what `earlierResult` was even for.

Reply:
That strategy won't work because we need to be able to evaluate nested iframes correctly - when an iframe is contained in another iframe that should pass, but the inner iframe does not pass on its own, we don't know until we've iterated sequentially through that the inner iframe should pass and thus be considered part of the set that should be matched against. 

Consider this (which is represented in the full integration tests as region-pass-nested-iframe.html): 
```
<main>
  <iframe id='outer' srcdoc="<iframe id='inner' srcdoc='<p>hello</p>'></iframe></iframe>
</main>
```

Here's the code I was working on, which has the problem described above, if you want to play around with it yourself.
```js
function regionAfter(results) {
  const passingFrameAncestries = results.filter(
    r => r.result && r.data.isIframe
  ).map(r => r.node.ancestry);

  results.forEach((r, index) => {
    // this element failed the check, and it's in an iframe
    if (!r.result && r.node.ancestry.length > 1) {
      // we need to see if it's actually contained in an iframe that passed the check
      const frameAncestry = r.node.ancestry.slice(0, -1);

      for (const frame of passingFrameAncestries) {
        if (matchAncestry(frameAncestry, frame)) {
          r.result = true;
          break;
        }
      }
    }
  });

  // iframe elements should always pass
  results.forEach(r => {
    if (!r.result && r.data.isIframe) {
      r.result = true;
    }
  });
  return results;
}
```